src/clients/AcrossConfigStoreClient/ConfigStoreParsingUtilities/ConfigStoreParsingUtilities.test.ts

```diff
@@ -0,0 +1,233 @@
+import { isUBAOnChainConfig } from "./ConfigStoreParsingUtilities";
+
+const validConfigStore: { reason: string; value: unknown }[] = [
+  {
+    reason: "valid config",
+    value: {
+      alpha: {
+        default: "23",
+      },
+      omega: {
+        default: {
+          cutoff: [],
+          value: [],
+        },
+      },
+      gamma: {
+        default: {
+          cutoff: [],
+          value: [],
+        },
+      },
+      rebalance: {
+        default: {
+          threshold_lower: "3",
+          threshold_upper: "3",
+          target_lower: "3",
+          target_upper: "3",
+        },
+        "1-1": {
+          threshold_lower: "3",
+          threshold_upper: "3",
+          target_lower: "3",
+          target_upper: "3",
+        },
+      },
+    },
+  },
+];
+const invalidConfigStore: { reason: string; value: unknown }[] = [
+  { reason: "Empty object", value: {} },
+  { reason: "Undefined", value: undefined },
+  { reason: "Null object", value: null },
+  { reason: "Primitive number", value: 1 },
+  { reason: "Primitive string", value: "string" },
+  { reason: "Empty array", value: [] },
+  { reason: "Array with single invalid config", value: [{}] },
+  { reason: "Array with single undefined config", value: [undefined] },
+  { reason: "Array with single nullified config", value: [null] },
+  { reason: "Array with single primitive number", value: [0] },
+  { reason: "Array with single string ", value: ["string"] },
+  {
+    reason: "Invalid config with rebalance not having a default",
+    value: {
+      alpha: {
+        default: "23",
+      },
+      omega: {
+        default: {
+          cutoff: [],
+          value: [],
+        },
+      },
+      gamma: {
+        default: {
+          cutoff: [],
+          value: [],
+        },
+      },
+      rebalance: {
+        defaulting: {
+          threshold_lower: "3",
+          threshold_upper: "3",
+          target_lower: "3",
+          target_upper: "3",
+        },
+      },
+    },
+  },
+  {
+    reason: "Invalid config with rebalance having a key that isn't '{number}-{number}'",
+    value: {
+      alpha: {
+        default: "23",
+      },
+      omega: {
+        default: {
+          cutoff: [],
+          value: [],
+        },
+      },
+      gamma: {
+        default: {
+          cutoff: [],
+          value: [],
+        },
+      },
+      rebalance: {
+        default: {
+          threshold_lower: "3",
+          threshold_upper: "3",
+          target_lower: "3",
+          target_upper: "3",
+        },
+        "1-": {
+          threshold_lower: "3",
+          threshold_upper: "3",
+          target_lower: "3",
+          target_upper: "3",
+        },
+      },
+    },
+  },
+  {
+    reason: "Invalid config with rebalance having a key that isn't '{number}-{number}'",
+
+    value: {
+      alpha: {
+        default: "23",
```

Comment:
What if `default` is empty, is that a case we should handle?

Reply:
Happy to enforce it! do you think this case will appear often?

Reply:
Alternatively, since we have a fixed default that will be hardcoded, it shouldn't necessarily be required. Following that logic, we may not need default at all since we have a protocol default.

Reply:
I would agree it shouldn't be required if we have a fixed default